### PR TITLE
[FEAT] Add training augmentation via MixUp, Audio Preprocessing Polishes, LLM2Vec PCA

### PR DESCRIPTION
**Pull request by HansQueja**
## Summary
Added MixUp augmentation to the MLP training pipeline for improved generalization on embedded features. Updated the training loop to handle MixUp when mixup_alpha > 0 in the configuration. Added PCA implementation for LLM2Vec for more general vectors. Polished audio preprocessing to add augmentation and variability.

### Changes Made
- [x] Implemented `mixup` and `mixup_loss` methods in MLPModel.
- [x] Updated `MLPClassifier.train()` to use MixUp augmentation when enabled in config.
- [x] Added config parameter `mixup_alpha` to control MixUp strength.
- [x] Refactored training loop to support MixUp without breaking existing functionality.
- [x] Added flexibility on reading audio via Librosa.
- [x] Added randomization on cropping/padding audio.
- [x] Added gaussian noise on waveform before returning.
- [x] Added preprocessing documentation inside `docs/`.
- [x] Adjusted and added comments. 

### Notes
- MixUp is applied only during training. Other methods such as prediction and validation should be unchanged.
- Future PRs can integrate MixUp with other data augmentation strategies for further performance improvements.
- One thing that I got to search about more is that saving preprocessed songs and then re-loading it could actually introduce artifacts. With that in mind, I opted to just upload the raw files, and let the Google Colab runs preprocess it and directly return it to the feature extractors. It might help to reduce tendencies of overfitting.

### CODE REVIEW

File: src/preprocessing/audio_preprocessor.py [hunk 1]
```diff
@@ -109,32 +120,40 @@ def resample_audio(self, original_sr, waveform):
         return waveform
         
 
-    def pad_trim(self, waveform):
+    def pad_trim(self, waveform, random_crop=False):
```
Comment (sykeruzn) [inline src/preprocessing/audio_preprocessor.py:123]: Clarification: This introduces random cropping/padding but still ensures the input is in 120s?

File: src/preprocessing/audio_preprocessor.py
Comment (sykeruzn) [inline src/preprocessing/audio_preprocessor.py:1]: So far, the audio preprocessor flow is still clear to me. I just have a few clarifications which I commented in the specific lines. Nice refinement on docstrings/comments to clarify the output of preprocessor which is normalized waveform, since this is also the expected input of SpecTTTra pipeline. Good job on adding the Gaussian noise for training too!

File: docs/preprocessing.md
Comment (sykeruzn) [inline docs/preprocessing.md:1]: Thank you for this! This is a really good help to understand the code structure and respective pipelines. I can adapt this for the SpecTTTra pipeline documentation too.

**Review by sykeruzn — APPROVED** [review]
All the changes looks good to me. I just have specific comments of clarification in some of the code files but overall, the structure and flow are clear to me. Thank you for the updates!
